Update issue and PR templates, add config, stale bot not

sagetherage opened the pull request:
### What does this PR do?
Uses GitHub templates for [Bug] and [Feature Request] adds labels, easier to find by syntax search, adds syntax to PR template `Fixes:` to close issues linked with PRs as stated in the Contributing Guidelines, plus adds a config file with links for questions and discussions.

### What issues does this PR fix or reference?
Fixes: #55853

### Tests written?
No

### Commits signed with GPG?
Yes

sagetherage committed:
Update issue templates

Adding GitHub templates for bug and feature request as these will automatically enforce the title and labels

sagetherage committed:
Merge pull request #1 from sagetherage/sagetherage-issue-temps

Update issue templates

sagetherage committed:
Creating config file for external site links

sagetherage committed:
Merge pull request #2 from sagetherage/sagetherage-issue-temp-config

Creating config file for external site links

sagetherage committed:
Update issue templates

sagetherage committed:
Update PR template 

adding syntax fixes

sagetherage committed:
Merge pull request #3 from sagetherage/sagetherage-PR-temp-syntax

Update PR template

Commit:
Signed commit

Commit:
Merge branch 'master' of github.com:sagetherage/salt

waynew commented on .github/ISSUE_TEMPLATE/config.yml:
```diff
@@ -0,0 +1,11 @@
+  - name: Salt-Users Forum
+    url: https://groups.google.com/forum/#!forum/salt-users
+    about: Please ask and answer questions here.
+     - name: Salt on Freenod 
```
```suggestion
     - name: Salt on Freenode
```

sagetherage replied:
@waynew ok all fixed thank you!

waynew reviewed (changes requested):
Found a missing `e` :upside_down_face:

sagetherage committed:
Update .github/ISSUE_TEMPLATE/config.yml

fix misspelling

Co-Authored-By: Wayne Werner <waynejwerner@gmail.com>

sagetherage committed:
Update config to include security

Added security email for contact links

sagetherage committed:
Update feature_request temp

Added wording that if this is an addition or big change to use the SEP process and link

max-arnold commented:
May I suggest adding the following snippet to the versions report section to make it collapsed by default?

    <details><summary>salt --versions-report</summary>
    
    ```
    PASTE HERE
    ```
    </details>

Another suggestion is to consider adding the `Docs & changelog written?` section to the PR template (in order to improve the docs coverage).

sagetherage committed:
Merge branch 'master' into master

sagetherage committed:
update bug_report-1

Make Versions Report section collapse

sagetherage committed:
add docs and changelog

add docs and changelog question to PR template

sagetherage committed:
update stale no close

update configuration to not close issues

bryceml commented:
https://github.com/saltstack/salt/pull/56075

could probably be taken care of on this pr and that one closed.

sagetherage committed:
update pr_temp master in URL-2

update PR template to include the correct URL to be master not latest

sagetherage commented:
I guess I decided @dwoz when you review this for merging, please comment.

waynew commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,23 +1,27 @@
 
+### Docs & changelog written?
+Yes/No 
+
```
Maybe do:

```suggestion
- [ ] Docs
- [ ] Changelog

```

waynew reviewed (changes requested):
I think this will render the GFM checkbox?

sagetherage committed:
Update .github/PULL_REQUEST_TEMPLATE.md

Co-Authored-By: Wayne Werner <waynejwerner@gmail.com>

sagetherage committed:
Merge branch 'master' into master

sagetherage committed:
Merge branch 'master' into master

sagetherage committed:
Merge branch 'master' into master
